File: src/test/java/com/github/kayjamlang/tests/expressions/callcreate/CallOrCreateExpressionZeroArgsTest.java
package com.github.kayjamlang.tests.expressions.callcreate;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.CallOrCreateExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class CallOrCreateExpressionZeroArgsTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("concat()"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(CallOrCreateExpression.class, expression.getClass());

        CallOrCreateExpression callOrCreateExpression = (CallOrCreateExpression) expression;
        assertEquals("concat", callOrCreateExpression.functionName);
        assertEquals(0, callOrCreateExpression.arguments.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/callcreate/CallOrCreateExpressionZeroArgsTest.scala
package com.github.kayjamlang.tests.expressions.callcreate

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.CallOrCreateExpression
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object CallOrCreateExpressionZeroArgsTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("concat()"))
  }
}

class CallOrCreateExpressionZeroArgsTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = CallOrCreateExpressionZeroArgsTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[CallOrCreateExpression], expression getClass)
    val callOrCreateExpression = expression.asInstanceOf[CallOrCreateExpression]
    assertEquals("concat", callOrCreateExpression.name)
    assertEquals(0, callOrCreateExpression.arguments size)
  }
}